other/upgrade_2-0_postgresql.sql

```diff
@@ -918,192 +918,67 @@ if (!isset($modSettings['attachment_thumb_png']))
 ---#
 
 /******************************************************************************/
---- Installing new default theme...
+--- Cleaning up after old themes...
 /******************************************************************************/
 
----# Installing theme settings...
+---# Checking for "babylon" and removing it if necessary...
 ---{
-// This is Grudge's secret "I'm not a developer" theme install code - keep this quiet ;)
-
-// Firstly, I'm going out of my way to not do this twice!
-if ((!isset($modSettings['smfVersion']) || $modSettings['smfVersion'] <= '2.0 RC2') && empty($modSettings['dont_repeat_theme_core']))
+// Do they have "babylon" installed?
+if (file_exists($GLOBALS['boarddir'] . '/Themes/babylon'))
 {
-	// Check it's not already here, just in case.
+	$babylon_dir = smf_mysql_real_escape_string($GLOBALS['boarddir'] . '/Themes/babylon');
 	$theme_request = upgrade_query("
-		SELECT id_theme
+		SELECT ID_THEME
 		FROM {$db_prefix}themes
 		WHERE variable = 'theme_dir'
-			AND value LIKE '%core'");
-	// Only do the upgrade if it doesn't find the theme already.
-	if ($smcFunc['db_num_rows']($theme_request) == 0)
+			AND value ='$babylon_dir'");
+
+	// Don't do anything if this theme is already uninstalled
+	if (smf_mysql_num_rows($theme_request) == 1)
```

Comment:
you are here in the postgre file ``smf_mysql_num_rows`` didn't work here.
You would make your life easier when you use the $smcFunc['db_*'] since they exists in both worlds.

Reply:
Hmmm...  All of the upgrade-helper.php "smf_" functions are mysql specific...  Let me think about this...

Reply:
@albertlast - I must admit I'm struggling with how far to take this.  

In the past, the steps were rewritten for pg vs mysql - hence there are even different files for upgrade_2-0_postgresql.sql vs upgrade_2-0_mysql.sql.   And they are very different...  

Question:  If we use smcFuncs throughout... can we get to the point where we don't even need different files, e.g., upgrade_2-0_postgresql.sql vs upgrade_2-0_mysql.sql?  Or, are there functions that cannot be abstracted using smcFunc needed in the upgrade?  

Do we "smcFunc" EVERYTHING?   Do we "smcFunc" only a little?  Or do we continue to rewrite the steps for pg?

Reply:
In general i don't understand why you add smf*mysql* function.

No the upgrade process is complex enough.

I would only not add any smf*mysql* function

Reply:
Those functions are heavily used throughout the 2.0 & 2.1 upgrade scripts already...  

Looking at the original 1.x scripts, they were mostly straight-up sql & very simple php.  A custom upgrade_query() that allowed for special treatment of certain sql errors was sufficient.  The 2.x scripts, however, require much more logic to migrate data & implement data fixes.  So there is a need to use functions beyond upgrade_query().  

While doing so, it appears the intent was to _avoid_ smcFuncs.  I believe the reason for this was initial concern was that the upgrader is 'pulling itself up by its own bootstraps', meaning, that SMF may not be fully configured to run properly under the target version yet.  So the workaround was to either use direct mysqli calls (in 2.0.x) or to use the smf_ calls in the upgrade_helper (in 2.1) - which, in turn, used direct mysqli calls...  

However we got here, the problem is that we cannot re-use logic across pg & mysql in the upgrader today...  

I'll experiment a bit to see how big a deal it is to use smcFuncs...

Reply:
upgrade_query() already uses $smcFunc.

https://github.com/SimpleMachines/SMF2.1/blob/4b6c2b145369a171055e3d2d4f90408e445a0f41/other/upgrade.php#L2120

Reply:
Yep, for db_query.  

The question is whether we should modify all of the upgrade*.sql files to using it.  None of them work that way now.

Reply:
eh...

It woulds need much testing, which translates to more delay/dev time. Maybe a bigger hammer should be put away.

Reply:
the problem is not the existing mysql function,
the problem is that you add some,
instead of using the newer api.

Reply:
What added function are you talking about?

Reply:
He means that you should use $smcFunc['db_num_rows'] instead of smf_mysql_num_rows.

Reply:
@live627 - My broader concern is that the 2.0 .sql files (e.g., in the 2.0.15 upgrade)  and the 2.1 .sql files (visible here on github) have drifted considerably.  This is not good, and I fear that forums may have different fixes applied (_or worse, not applied..._).  I believe they really should match - preferably byte-for-byte.  _One_ of the differences is that right now, 2.0 uses primarily direct mysqli calls (with a couple smcFuncs), where 2.1 mainly uses all of these smf_calls from upgrade_helper.php (which makes mysqli calls...).  

At some point - sooner rather than later - they should be brought back in sync in some agreed-upon fashion.  When doing so, I agree with albertlast that we should use the existing smcFuncs in both branches, and get rid of direct mysqli calls.

Reply:
@frandominguez03 - that's not new by any stretch.  They've been used throughout the 2.1 .sql scripts for years...

Reply:
All - What I'm going to do here is do targeted fixes for albertlast's comments.

We will look at the proper sync'ing of the files as a separate issue/PR.  It's important to get 1.x working again.

Reply:
Converted to smcFunc